appinventor/components/src/com/google/appinventor/components/runtime/Regression.java

```diff
@@ -0,0 +1,125 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2022-2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime;
+
+import com.google.appinventor.components.annotations.*;
+import com.google.appinventor.components.common.*;
+import com.google.appinventor.components.runtime.util.YailDictionary;
+import com.google.appinventor.components.runtime.util.YailList;
+import gnu.lists.LList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A data science component to apply different regression models
+ * The component requires a data source to apply the model on
+ */
+@DesignerComponent(version = YaVersion.REGRESSION_COMPONENT_VERSION,
+    description = "A component that contains regression models",
+    category = ComponentCategory.DATASCIENCE,
+    iconName = "images/web.png",
+    nonVisible = true)
+@SimpleObject
+@SuppressWarnings("checkstyle:JavadocParagraph")
+public final class Regression extends DataCollection {
+
+  /**
+   * Creates a new Regression component.
+   */
+  public Regression(ComponentContainer container) {
+    super();
+    // Construct default dataFileColumns list with 2 entries
+    dataFileColumns = Arrays.asList("", "");
+    sheetsColumns = Arrays.asList("", "");
+    webColumns = Arrays.asList("", ""); // Construct default webColumns list with 2 entries
+  }
+
+  /**
+   * Calculates the line of best fit
+   *
+   * @param xEntries - the list of x values
+   * @param yEntries - the list of y values
+   * @return list. 1st element of the list is the slope, 2nd element is the intercept, 3rd correlation coefficient, 4th element is the line of best fit prediction values
+   */
+  public static YailDictionary ComputeLineOfBestFit(final YailList xEntries, final YailList yEntries) {
+    LList xValues = (LList) xEntries.getCdr();
+    List<Double> x = castToDouble(xValues);
+
+    LList yValues = (LList) yEntries.getCdr();
+    List<Double> y = castToDouble(yValues);
+
+    if (xValues.size() != yValues.size())
+      throw new IllegalStateException("Must have equal X and Y data points");
```

Comment:
Need to also check that there are at least 2 points in the dataset as well.
Code style: wrap the body of the if statement in curly braces.